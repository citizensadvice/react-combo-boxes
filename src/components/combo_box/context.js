import { createContext } from 'react';
import { visuallyHiddenClassName } from '../../constants/visually_hidden_class_name';

export const Context = createContext({
  visuallyHiddenClassName,
  dispatch: null,
});
