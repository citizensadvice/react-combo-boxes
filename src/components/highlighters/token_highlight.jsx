import React from 'react';
import { HighlightValue } from '../highlight_value';
import { tokenHighlighter } from '../../highlighters/token_highlighter';

export function TokenHighlight(props) {
  return (
    <HighlightValue
      highlight={tokenHighlighter}
      {...props}
    />
  );
}
