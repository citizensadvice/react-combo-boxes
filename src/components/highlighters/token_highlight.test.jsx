import React from 'react';
import { render } from '@testing-library/react';
import { TokenHighlight } from './token_highlight';
import { Context } from '../../context';

function TestHighlight({ children, value, inverse, ...props }) {
  return (
    <Context.Provider value={{ ...props, props: { value, visuallyHiddenClassName: 'sr-only' } }}>
      <TokenHighlight inverse={inverse}>
        {children}
      </TokenHighlight>
    </Context.Provider>
  );
}

it('does not highlight with no children', () => {
  const { container } = render((
    <TestHighlight search="bar" />
  ));

  expect(container).toContainHTML('<div></div>');
});

it('does not highlight with no search children', () => {
  const { container } = render((
    <TestHighlight search={null}>
      foo
    </TestHighlight>
  ));

  expect(container).toContainHTML('<div>foo</div>');
});

it('does not highlight no match', () => {
  const { container } = render((
    <TestHighlight search="bar">
      foo foobar
    </TestHighlight>
  ));

  expect(container).toContainHTML('<div>foo foobar</div>');
});

it('highlights all tokens', () => {
  const { container } = render((
    <TestHighlight search="bar">
      foo bar foo bar
    </TestHighlight>
  ));

  expect(container).toMatchSnapshot();
});

it('highlights the start of tokens', () => {
  const { container } = render((
    <TestHighlight search="bar">
      &quot;barfoo&quot;
    </TestHighlight>
  ));

  expect(container).toMatchSnapshot();
});

it('highlights existing value', () => {
  const { container } = render((
    <TestHighlight value={{ label: 'bar' }}>
      foo bar foo bar
    </TestHighlight>
  ));

  expect(container).toMatchSnapshot();
});

it('inverses the highlight', () => {
  const { container } = render((
    <TestHighlight search="bar" inverse>
      foo bar foo bar
    </TestHighlight>
  ));

  expect(container).toMatchSnapshot();
});
