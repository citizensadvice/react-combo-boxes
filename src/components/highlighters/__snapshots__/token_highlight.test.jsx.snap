// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`highlights all tokens 1`] = `
<div>
  <span
    class="sr-only"
  >
    foo bar foo bar
  </span>
  <span
    aria-hidden="true"
  >
    foo 
    <mark>
      bar
    </mark>
     foo 
    <mark>
      bar
    </mark>
  </span>
</div>
`;

exports[`highlights existing value 1`] = `
<div>
  <span
    class="sr-only"
  >
    foo bar foo bar
  </span>
  <span
    aria-hidden="true"
  >
    foo 
    <mark>
      bar
    </mark>
     foo 
    <mark>
      bar
    </mark>
  </span>
</div>
`;

exports[`highlights the start of tokens 1`] = `
<div>
  <span
    class="sr-only"
  >
    "barfoo"
  </span>
  <span
    aria-hidden="true"
  >
    "
    <mark>
      bar
    </mark>
    foo"
  </span>
</div>
`;

exports[`inverses the highlight 1`] = `
<div>
  <span
    class="sr-only"
  >
    foo bar foo bar
  </span>
  <span
    aria-hidden="true"
  >
    <mark>
      foo 
    </mark>
    bar
    <mark>
       foo 
    </mark>
    bar
  </span>
</div>
`;
